src/extensions/default/RelatedFiles/main.js

```diff
@@ -0,0 +1,202 @@
+    * @returns {Boolean} true if the url matches regular expression for remote url.
+    **/
+	function isRemoteURL(tagurl) {
+		var pattern = RegExp('^((http|https|ftp)?:?\/\/)');
+        return pattern.test(tagurl); 
+	} 
+    
+    /**
+    * Create a SearchResult object which has path information about the local file.
+    * @params {string} tagurl The url from which file path information is to be extracted.
+    * @returns {Object} SearchResult object which has file path information.
+    **/
+    function getLinkFromLocalURL(tagurl) {
+        var searchResult = new StringMatch.SearchResult (tagurl);
+        var file = Filesystem.getFileForPath(tagurl);
+        searchResult.label = file._name;
+        searchResult.fullPath = file._path;
+        searchResult.stringRanges = [{
+            text: file._path,
+            matched: false,
+            includesLastSegment: true
+        }];
+        return searchResult;
+    }
+    
+    /**
+    * Create a SearchResult object which has path information about the remote file url.
+    * @params {string} tagurl The url from which file path information is to be extracted.
+    * @returns {Object} SearchResult object which has remote link path information.
+    **/
+    function getLinkFromRemoteURL (tagurl) {
+        var regex = /:\/\/(.[^/]+)/;
+        var url = regex.exec(tagurl);
+        if (url && url[1]) {
+            var searchResult = new StringMatch.SearchResult(tagurl);
+            searchResult.label =   url[1];
+            searchResult.fullPath = tagurl;
+            searchResult.stringRanges = [{
+                text: tagurl,
+                matched: false,
+                includesLastSegment: false,
+                includesFirstSegment: true
+            }];
+            return searchResult;
+        }
+        return null;
+    }
+
+    /**
+    * Create abosulte path for a file from relative path.
+    * @params {string} base Path of the current document.
+    * @params {string} relative Relative path of a file which is to be converted to absolute path.
+    * @returns {string} Absolute path of the given relative path with calculated wrt base.
+    **/
+    function getAbsoultePath(base, relative) {
+        var baseUrl = base.split("/"),
+            relativeUrl = relative.split("/");
+        baseUrl.pop();
+    
+        for(var i = 0; i < relativeUrl.length; i++) {
+            if (relativeUrl[i] === ".") {
+                continue;
+            }
+            if (relativeUrl[i] === "..") {
+                baseUrl.pop();
+            }
+            else {
+                baseUrl.push(relativeUrl[i]);
+            }
+        }
+        return baseUrl.join("/");
+    }
+    
+    function RelatedFiles() {
+        
+    }
+    
+    /**
+    * Gets list of realted files.
+    * @returns {List} List of realted files.
+    **/
+    RelatedFiles.prototype.getRelatedFiles = function() {        
```

Comment:
Lets keep this signature uniform which can be leveraged later by other extensions as well. IMO, this should be a `promise` returning function.